templates/packer/ansible/roles/common/files/worker-initialization.service

```diff
@@ -21,6 +21,7 @@ ExecStart=/usr/local/libexec/worker-initialization-scripts/get_oci_creds.sh
 ExecStart=/usr/local/libexec/worker-initialization-scripts/get_pulp_creds.sh
 ExecStart=/usr/local/libexec/worker-initialization-scripts/get_ldap_sa_mtls_creds.sh
 ExecStart=/usr/local/libexec/worker-initialization-scripts/worker_service.sh
+ExecStopPost=/usr/local/libexec/worker-initialization-scripts/on_failure.sh
```

Comment:
Are you sure that this won't be run on successful runs? `man systemd.service` kinda implies that services with `Type=oneshot` are considered dead if they don't have `RemainAfterExit=` set.

> Behavior of oneshot is similar to simple; however, the service manager will consider the unit up after the main process exits. It will then start follow-up units.  RemainAfterExit= is particularly useful for this type of service.  Type=oneshot is the implied default if neither Type= nor
               ExecStart= are specified. Note that if this option is used without RemainAfterExit= the service will never enter "active" unit state, but will directly transition from "activating" to "deactivating" or "dead", since no process is configured that shall run continuously. In particular this means
               that after a service of this type ran (and which has RemainAfterExit= not set) it will not show up as started afterwards, but as dead.

I suggest trying this locally. If you already did, please ignore me. :)

Reply:
egh no you're right
I think OnFailure with a separate service might be the way to go. Or checking the service_result.